test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/ExecutionBoundaries.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.test.util.bpmn.random;
+
+import java.util.Random;
+import org.agrona.collections.IntArrayList;
+
+/**
+ * An execution boundary is the point where automatic and non-automatic {@link
+ * ScheduledExecutionStep}'s meet each other. This class contains information about the existing
+ * execution boundaries in a {@link ExecutionPath}.
+ */
+final class ExecutionBoundaries {
+
+  private final IntArrayList boundaries = new IntArrayList();
```

Comment:
Nit (and kind of random :smile:): why use Agrona classes when performance isn't important over standard things?

Reply:
Because  I like it 😃 it is the only collection which avoids the boxing, but yeah is not necessary

Reply:
I'm surprised, since you usually prefer to use standard things ;)

But like I said, not a blocker - I would just use a normal list here since I don't care for performance, but it's also fine to use this. The only downside I can imagine is if we ever wanted to not use Agrona, this would increase how much we have to change, but...yeah, I doubt this will happen anytime soon :sweat_smile: